docs/getting-started/application/patterns.rst

```diff
@@ -53,122 +53,61 @@ The above policy controls access to the Expense model on the **record level**.
-    evaluation point within the **data access layer**.  This
-    ensures that the ``oso.is_allowed`` call is made when an expense is accessed, no
-    matter where that access occurs in the application.
-
-.. todo::
-
-   Would be nice to be able to say here: "see this guide for an
-   example of how to do this..."
-
 Model Level Access Control
 ==========================
 
-Sometimes, access control does not rely on properties of the primary data.  This
-type of access control is called **model level**.
+Sometimes, access control does not depend on the properties of the resource but just
+on it's type. This is called **model level** can be used for coarse level authorization.
+This is useful if you want to restrict access to a whole page if the user cannot see any
+of the resources of the kind the page is about. Since the rules depend only on the type
+of the model and not on the data it makes it easier to add these checks upstream in
+middleware.
 
 .. code-block:: polar
     :caption: :fa:`oso`
 
-    allow(actor, "view", "expense") if actor.role = "accountant";
-    allow(actor, "modify", "team") if actor.role = "hr_admin";
-    allow(actor, "modify", "project") if actor.role = "hr_admin";
-    allow(actor, "modify", "organization") if actor.role = "hr_admin";
+    allow(actor, "view", "Expense") if actor.role = "accountant";
+    allow(actor, "modify", "Team") if actor.role = "hr_admin";
+    allow(actor, "modify", "Project") if actor.role = "hr_admin";
+    allow(actor, "modify", "Organization") if actor.role = "hr_admin";
 
 This brief policy shows an example of model level access control:
 
 - An accountant can view expenses.
 - HR admins can modify teams, projects, and organizations.
 
-Notice that this policy does not rely on any **primary authorization data**.
-Therefore it can be evaluated either before or after the primary data fetch.
-Here's what it would look like before:
+In this example we are representing the model type as a string and writing the rule
+over that.
 
 .. code-block:: python
     :caption: :fab:`python`
 
-    def get_expense(user, id):
+    def expense_page(user, id):
         # See if the user is allowed to access expenses at all.
-        if oso.is_allowed(user, "view", "expense"):
-            expense = db.fetch(
-                "SELECT * FROM expenses WHERE id = %", expense_id)
-            # Process request
-        else:
-            # Not authorized
+        if not oso.is_allowed(user, "view", "Expense"):
             return NotAuthorizedResponse()
+        # Process request
 
-.. note::
-
-    You may have noticed that this policy still accesses **actor data**.  This
-    is fine, since usually this data will be fetched prior to authorization as
-    part of the authentication flow.
+Model level is also sometimes a good way to write rules about creating new resources.
+You can write a rule saying if a user can create an Expense or not.
```

Comment:
It would be good to have more explanation here of why model-level access control is good for this.